### PR TITLE
Keep link format active on edit.

### LINKED ISSUE #25946: Click on "Open in new tab" control causes hyperlink creation UI popover to close
**Describe the bug**
When creating a link on a portion of text in a paragraph block you may wish for the link to open in a new tab. There is a control in the hyperlink creation popover UI which allows you to toggle this feature on/off. However, when it is clicked the hyperlink creation UI popover closes. 

This is undesirable as you may wish to toggle this feature on/off several times or you may have made a mistake. In this case the fact that the UI disappears each time a change is made is quite frustrating. I assume this is due to a re-render of the `LinkControl` component but I could be wrong.

**To reproduce**
1. Create new Post
2. Add some text in a paragraph block
3. Highlight some of the text and create a link.
4. Now use the toggle control for "Open in new tab".
5. Notice how clicking/toggling the control will cause the link UI to disappear.

**Expected behavior**
The link creation UI should not close when the toggle control is interacted with. It should remain open until it is dismissed.

**Screenshots**

Screen Capture on 2020-10-08 at 09-27-05

**Editor version (please complete the following information):**
- WordPress version: 5.5.1
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Plugin
- If the Gutenberg plugin is installed, which version is it? 9.9.1

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome
 - Version 85.0.4183.121 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by willRicard**
## Description

Clearing the `activeFormats` list on the link value caused the link edit
popover to close.

Reported by Dave Smith <getdavemail@gmail.com>.
Fixes #25946.
## How has this been tested?

Automatic test pipeline.

## Types of changes

Bug fix
## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by guarani — APPROVED** [review]
I left a comment [here](https://github.com/WordPress/gutenberg/issues/25946#issuecomment-712127916) just to note that although I could reproduce this issue, it only happened while editing an existing link — not while adding a new link.
Is this what you also saw @willRicard?

I gave it a quick test on Gutenberg and the fix appears to be working. I then gave it a test on [Mobile Gutenberg](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/getting-started-native-mobile.md) (where this bug wasn't present) to check for regressions and everything appears to be working as expected.

I think [someone from the core team will have to review this](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#merging-pull-requests) before it can be merged.

**Comment by willRicard** [comment]
Thanks @guarani the issue only happened on existing links as you say. I'll give it a go on mobile as well :)

**Comment by willRicard** [comment]
Hello @wordpress/gutenberg-core, I just rebased my PR. Please note the failing e2e tests also occurs on `master`.

**Comment by youknowriad** [comment]
Sorry I missed this PR. It seems you didn't ping the team properly :) Mind rebasing again? 

Also, do we know the original purpose of that removed line?

**Comment by ellatrix** [comment]
Was added for #7980. Also, it doesn't make sense to collapse the linked selection and then leave the caret in the link... Either we should keep the selection or move it after the link.

**Comment by ellatrix** [comment]
Suggesting close for this PR.

**Comment by willRicard** [comment]
Sadly yes it must go, thanks for following up on this 👍

**Pull request closed without merging**
